pac4j-core/src/main/java/org/pac4j/core/profile/ProfileHelper.java

```diff
@@ -56,55 +58,71 @@ public static boolean isTypedIdOf(final String id, final Class<? extends UserPro
-                            completeName = "org.pac4j.cas.profile." + className;
-                        } else if ("HttpTGTProfile".equals(className)) {
-                            completeName = "org.pac4j.cas.profile.HttpTGTProfile";
-                        } else if ("SAML2Profile".equals(className)) {
-                            completeName = "org.pac4j.saml.profile.SAML2Profile";
-                        } else if ("HttpProfile".equals(className)) {
-                            completeName = "org.pac4j.http.profile.HttpProfile";
-                        } else if ("OidcProfile".equals(className)) {
-                            completeName = "org.pac4j.oidc.profile.OidcProfile";
-                        } else if ("LdapProfile".equals(className)) {
-                            completeName = "org.pac4j.ldap.profile.LdapProfile";
-                        } else if ("DbProfile".equals(className)) {
-                            completeName = "org.pac4j.sql.profile.DbProfile";
-                        } else if ("MongoProfile".equals(className)) {
-                            completeName = "org.pac4j.mongo.profile.MongoProfile";
-                        } else if ("YahooOpenIdProfile".equals(className)) {
-                        	completeName = "org.pac4j.openid.profile.yahoo.YahooOpenIdProfile";
-                        } else if ("GaeUserServiceProfile".equals(className)) {
-                            completeName = "org.pac4j.gae.profile.GaeUserServiceProfile";
-                        } else if ("StormpathProfile".equals(className)) {
-                            completeName = "org.pac4j.stormpath.profile.StormpathProfile";
-                        } else {
-                            final String packageName = className.substring(0, className.length() - 7).toLowerCase();
-                            completeName = "org.pac4j.oauth.profile." + packageName + "." + className;
-                        }
-                        @SuppressWarnings("unchecked")
-                        final Constructor<? extends UserProfile> constructor = (Constructor<? extends UserProfile>) Class
-                                .forName(completeName).getDeclaredConstructor();
-                        final UserProfile userProfile = constructor.newInstance();
-                        userProfile.build(typedId, attributes);
-                        logger.debug("userProfile built : {}", userProfile);
-                        return userProfile;
-                    } catch (final Exception e) {
-                        logger.error("Cannot build instance", e);
-                    }
-                }
+        if (CommonHelper.isBlank(typedId)) {
+            return null;
+        }
+
+        logger.info("Building user profile based on typedId {}", typedId);
+        try {
+            if (!typedId.contains("#")) {
+                final String completeName = determineProfileClassByName("HttpProfile");
```

Comment:
"the JwtAuthenticator must capture the subject field from the claims, and handle the separator there, right?": yes, you get it right.

Yes, it makes sense to create a specific `JwtProfile` (inheriting from `HttpProfile`) when we get a JWT token which was not previously a pac4j user profile. Indeed, you need to check the subject to see if there is a `#` otherwise you'll provide `JwtProfile.class.getSimpleName() + UserProfile.SEPARATOR`.